csrc/scheduler/utils.cpp

```diff
@@ -2666,6 +2666,83 @@ int64_t reorderDevicesToOuter(TensorView* tv) {
   return (int64_t)old2new.size();
 }
 
+void reorderTensorLike(
+    TensorView* target_tv,
+    const std::vector<IterDomain*>& ref) {
+  const auto& tv_loop_domain = target_tv->getLoopDomain();
+
+  IdModel id_model(target_tv->fusion(), /*build_graphs=*/false);
+  const auto& graph = id_model.buildBroadcastGraph();
+
+  ValGroups target_groups = graph.toGroups(tv_loop_domain);
+
+  ValGroups ref_groups = graph.toGroups(ref);
+
+  // Traverse from the reference to the target tv. The reference is
+  // not guaranteed to cover all loop IDs of target, so
+  // require_all_to_visited needs to be false
+  auto path = ValGraphBFS::getExprGroupsBetween(
+                  graph,
+                  ref_groups,
+                  target_groups,
+                  /*require_all_to_visited=*/false)
+                  .first;
+
+  // Traverse the expr path to create an ordered ID groups
+  std::deque<ValGroup> ordered_domain{
+      ref_groups.vector().begin(), ref_groups.vector().end()};
+
+  for (const auto& [expr_g, dir] : path) {
+    auto inputs = getInputsOfExpr(
+        expr_g, dir, ValGraphInputs(graph), ValGraphOutputs(graph));
+    auto outputs = getOutputsOfExpr(
+        expr_g, dir, ValGraphInputs(graph), ValGraphOutputs(graph));
+
+    // Inserts the outputs at the innermost position
+    std::deque<ValGroup>::iterator innermost_it = ordered_domain.end();
+    for (auto it = inputs.rbegin(); it != inputs.rend(); ++it) {
+      innermost_it =
+          std::find(ordered_domain.begin(), ordered_domain.end(), *it);
+      NVF_ERROR(innermost_it != ordered_domain.end());
```

Comment:
why is this iterator called `innermost_it` when it's breaking upon the first reference encountered?

Reply:
so it's the innermost inputs? but I thought the order of inputs aren't significant, but the position in ordered_domain is...

Reply:
Hmm, I don't remember why I had the loop here. It just needs to find the position of the innermost input in `ordered_domain`. It should be just:

```
std::deque<ValGroup>::iterator innermost_it = 
          std::find(ordered_domain.begin(), ordered_domain.end(), inputs.back());
NVF_ERROR(innermost_it != ordered_domain.end());
```